### DISCUSSION TITLE
Patching questions

### DISCUSSION THREAD

**Discussion by wmedernach** [Esri/arcgis-powershell-dsc#580] [Q&A] [answered]
I have some questions on patching a fresh install via DSC and the patching parameters.

- When setting ConfigData.DownloadPatches=true does it simply use the *.PatchesDir as the download location?
- I see there are currently ~50 patches for 10.9.1, so does it know what to install and the order based on the patch names?
- With ~50 patches available I'm not sure how you would populate the PatchInstallOrder parameter?
- I suppose a list of patches could be extracted from the ArcGIS Enterprise Patch Notification tool, but again the question would be is the listed order the correct install order?

Or, am I overthinking all of this? ;-)
![esri_patches](https://github.com/user-attachments/assets/4822dc36-a4d8-47d3-b9f7-0e6922eaa7b1)

**Comment by cameronkroeker** [comment, marked as answer]
Hi @wmedernach,

Here is a resource that may help answer some of your questions:

https://github.com/Esri/arcgis-powershell-dsc/wiki/Download-and-Install-patches-for-ArcGIS-Enterprise#option-1-download-and-install-all-available-patches-for-the-specified-version

The ArcGIS Module will download/install in the order of the release date of each patch (earliest to latest). ArcGIS Server 10.9.1 is a bit unique in that it requires the Setup Program Patch to be installed first, however the Setup Program patch wasn't released first. I believe the Machine Rename patch was released first then the setup program patch. So if no install order is specified it will attempt to install the first released patch which will fail, then it will install the Setup Program patch which will then allow all the other patches to install successfully. To workaround this you can either just re-run DSC again and it will install the patch that didn't originally install or specify the patch install order.

Thanks,
Cameron K.

**Reply by wmedernach** [reply]
@cameronkroeker, again, thanks so much for the reference and the extra details!!

One more question for confirmation. If I do want to provide the ordered list of patches to install, it looks like I need to provide the patch **file names**, and not the patch names as seen in the screen cap above from the Patch Notification tool.

**Reply by cameronkroeker** [reply]
Yes, that is correct. Here is an example:

```json
"PatchInstallOrder": [
   "ArcGIS-1091-S-SP-Patch.msp",
   "ArcGIS-1091-S-MR-PatchB.msp",
   "ArcGIS-1091-S-Log4j-PatchB.msp",
   "ArcGIS-1091-S-SEC2022U1-PatchB.msp",
   "ArcGIS-1091-S-SEC2022U2-Patch.msp",
   "ArcGIS-1091-S-MPS-Patch.msp",
   "ArcGIS-1091-S-UNDM1-Patch.msp",
   "ArcGIS-1091-S-UNDM2-Patch.msp",
   "ArcGIS-1091-S-K1-Patch.msp",
   "ArcGIS-1091-S-VRPG-Patch.msp",
   "ArcGIS-1091-S-SQ-Patch.msp",
   "ArcGIS-1091-S-UNDM3-Patch.msp",
   "ArcGIS-1091-S-GSQ-Patch.msp",
   "ArcGIS-1091-S-CPS-Patch.msp"
]
```
